Review example How to add custom buttons ( Actions ) to the lookup (#2)
DevExpress-Examples/XAF_how-to-add-custom-buttons-actions-to-the-lookup-and-popup-windows: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E466)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # XAF - How to add custom buttons ( Actions ) to the lookup and popup windows

Reviewer: ```suggestion
# XAF - Add Custom Buttons ( Actions ) to Lookup and Popup Windows
```

Is it really important to note that we add not only buttons but actions? From my perspective, buttons =  actions since all buttons implement actions. 

If you added this info to match a google query, I can suggest you to move this info from the header to the first sentence.

```suggestion
# XAF - Add Custom Buttons to Lookup and Popup Windows
```

I've also removed 'the' from the header to match this phrase to our [updated guideline](http://tw.devexpress.devx/language-guidelines/articles/#plural-countable-nouns)

> You can show custom buttons ([Actions](https://docs.devexpress.com/eXpressAppFramework/112622/ui-construction/controllers-and-actions/actions)) near the **New**, **Clear**, **OK**, and **Cancel** buttons in lookup and popup List Views. For this purpose, create a new View Controller with a [Simple Action](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.SimpleAction) and set its [`Category`](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.ActionBase.Category) property to `PopupActions`.

Reviewer: In the header, we stated this example adds custom button to the lookup and popup **windows**. In this sentence, we are talking about lookup and popup **List Views**. It confuses me. Are those elements the same?

I believe we should stick to a single definition (window or list view):

```suggestion
This example adds a new **Action In Popup** button to popup/lookup windows. Click on this button invokes a custom [action](https://docs.devexpress.com/eXpressAppFramework/112622/ui-construction/controllers-and-actions/actions) (display a sample message box). 

To display a custom button, create a new [ViewController](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ViewController) with a [Simple Action](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.SimpleAction) and set its [`Category`](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.ActionBase.Category) property to `PopupActions`.
```

Reviewer: I also removed the list of buttons where a user can add custom buttons. It seems that those are all buttons that can be displayed in this area of lookup/popup windows. Am I right, or there can be buttons that restrict to add new buttons to?

Author: ```suggestion
This example adds a new **Action In Popup** button to popup/lookup windows. Click on this button invokes a custom [Action](https://docs.devexpress.com/eXpressAppFramework/112622/ui-construction/controllers-and-actions/actions) (display a sample message box). 

To display a custom button, create a new [ViewController](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ViewController) with a [Simple Action](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.SimpleAction) and set its [`Category`](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.Actions.ActionBase.Category) property to `PopupActions`.
```

> This example adds a new **Action In Popup** Action to popup/lookup List Views. When a user clicks this Action, the application displays a sample message box.

Reviewer: ```suggestion

```

I'd remove that sentence since we have told this info in the first sentence.

Refer to the following topic for more information: [Add Actions to a Pop-up Window](https://docs.devexpress.com/eXpressAppFramework/112804/ui-construction/controllers-and-actions/add-actions-to-a-popup-window).

![image](https://user-images.githubusercontent.com/14300209/225921128-c14fc411-d304-4cb8-9d47-2080f2d1ec86.png)

## Files to Review

* [PopupListViewController.cs](./CS/EFCore/PopupActionEF/PopupActionEF.Module/Controllers/PopupListViewController.cs)

## Documentation

* [How to: Reorder an Action Container's Actions Collection](https://docs.devexpress.com/eXpressAppFramework/112815/ui-construction/controllers-and-actions/actions/how-to-reorder-an-action-containers-actions-collection)
* [How to: Place an Action in a Different Location](https://docs.devexpress.com/eXpressAppFramework/402145/ui-construction/controllers-and-actions/actions/how-to-place-an-action-in-a-different-location)

> * [Action Containers](https://docs.devexpress.com/eXpressAppFramework/112610/ui-construction/action-containers)

Reviewer: ```suggestion
* [Action Containers](https://docs.devexpress.com/eXpressAppFramework/112610/ui-construction/action-containers)
* [Actions](https://docs.devexpress.com/eXpressAppFramework/112622/ui-construction/controllers-and-actions/actions)
```

Reviewer: I'd add a link to the actions topic since we have linked it in our text. It's up to you
